pymc_experimental/model/transforms/autoreparam.py

```diff
@@ -174,15 +176,18 @@ def vip_reparam_node(
 ) -> Tuple[ModelDeterministic, ModelNamed]:
     if not isinstance(node.op, RandomVariable | SymbolicRandomVariable):
         raise TypeError("Op should be RandomVariable type")
-    rv = node.default_output()
-    try:
-        [rv_shape] = constant_fold([rv.shape])
-    except NotConstantValueError:
-        raise ValueError("Size should be static for autoreparametrization.")
+    _, size, *_ = node.inputs
+    eval_size = size.eval(mode="FAST_COMPILE")
+    if eval_size is not None:
```

Comment:
eval_size will never be `None`

Reply:
It can
<img width="807" alt="image" src="https://github.com/user-attachments/assets/5d17d6a1-3f7e-4000-9926-c401d244b512">

Reply:
Wait why are you evaling size instead of rv.shape?

Reply:
I was afraid I get the dependence on the original RV

Reply:
If you use eval you can only get a numerical output

Reply:
It does not seem to work, it complains on missing input error, probably the value var, I'm reverting back

Reply:
This is wrong max. If size is none the RV shape is implied by the parameters, it doesn't mean it is as scalar ()

Reply:
What missing input? There shouldn't be any inputs in the forward graph

Reply:
Value var should not ever show up in the forward graph

Reply:
Ah this is working on the fgraph...

Reply:
I have a dedicated method to update lambdas in the VIP class, but It is not handy after model transformations. It would be a good proposal to change that API later

Reply:
> So different question: can we create shared variable with data=np.zeros((1,)*rv.ndim) and only set correct shape later?
It should be possible, I assume that would mean I only know the shape after transformation is already done

Reply:
Do you need shape of transformed or original? If it is the transformed, you could request model.initial_point before starting the dispatch process?

Alternatively we should be able to use shape inference but may need to implement the `infer_shape` of the ModelVariable. The The helper `infer_static_shape` possibly followed by an eval should do the job and not complain about value variables because those shouldn't be in the graph after `infer_shape`

Reply:
Can you point me to which test failed when you tried `rv.shape.eval()`?

Reply:
this one: https://github.com/pymc-devs/pymc-experimental/actions/runs/10058239228/job/27800877978#step:5:1461

Reply:
> Do you need shape of transformed or original? If it is the transformed, you could request model.initial_point before starting the dispatch process?

This looks like an interesting solution

Reply:
> This looks like an interesting solution

```
model.initial_point()
```

Seems to give a point in the sampling space, I need the untransformed one